### PR TITLE
feat(module: eks-monitoring) Add NVIDIA gpu monitoring dashboards

### PR DESCRIPTION
**Pull request by lewinkedrs**
### What does this PR do?

This will add support for existing clusters with gpu-operator. The NVIDIA DCIM dashboards will be deployed and be connected to AMP.

Closes https://github.com/aws-observability/terraform-aws-observability-accelerator/issues/233

### CODE REVIEW

File: docs/eks/gpumon.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Monitoring NVIDIA GPU Workloads
+
+GPUs play an integral part in data intensive workloads. The base infrastructure module of the Observability Accelerator proivdes the ability to deploy the NVIDIA DCGM Exporter Dashboard.
```
Comment (bonclay7) [inline docs/eks/gpumon.md:3]: what base module are you referring to? [eks-monitoring module](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/modules/eks-monitoring)? maybe we should be specific 
also little typo on "provides"

File: docs/eks/gpumon.md
Comment (bonclay7) [inline docs/eks/gpumon.md:1]: maybe the file name should be explicit like `/docs/eks/gpu-monitoring.md` to help with better SEO?

File: docs/eks/gpumon.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+The dashboard utilizes metrics scraped from the `/metrics` endpoint that are exposed when running the nvidia gpu operator and NVSMI binary.
+
+!!!note
+    In order to make use of this dashboard, you will need to have a GPU backed EKS cluster and deploy the [GPU operator](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/amazon-eks.html)
+    The recommended way of deploying the GPU operator is the [Data on EKS Blueprint](https://github.com/aws-ia/terraform-aws-eks-data-addons/blob/main/nvidia-gpu-operator.tf)
+
+## Deployment
+
+This is enabled by default in the [base infrasturcture module](https://aws-observability.github.io/terraform-aws-observability-accelerator/eks/).
```
Comment (bonclay7) [inline docs/eks/gpumon.md:12]: maybe we should call it differently, like Managed open-source EKS monitoring? I gave it a shot in this PR #255 - read this https://github.com/aws-observability/terraform-aws-observability-accelerator/blob/49764dc834f3c38c0b8788f5713fe78c66b1fae0/docs/index.md

**Review by bonclay7 — COMMENTED** [review]
Nice! Please see my comments, need to setup NVIDIA to actually tests metrics collection now

File: docs/eks/gpumon.md (hunk 2 above)
Comment (lewinkedrs) [reply docs/eks/gpumon.md:12]: Call what differently? You mean from a documentation perspective the title or call the base eks-monitoring module something different? "Managed open-source EKS monitoring" seems like a mouthful. I think as long as there is no 'native' alternative module we can just go with eks-monitoring.

Comment (bonclay7) [reply docs/eks/gpumon.md]: just the doc name `[base infrasturcture module]`. I don't think we should rename the `eks-monitoring` module.

File: docs/eks/gpu-monitoring.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Monitoring NVIDIA GPU Workloads
+
+GPUs play an integral part in data intensive workloads. The base infrastructure module of the Observability Accelerator provides the ability to deploy the NVIDIA DCGM Exporter Dashboard.
```
Comment (bonclay7) [inline docs/eks/gpu-monitoring.md:3]: again `base infrastructure module` is not specific

Comment (lewinkedrs) [reply docs/eks/gpu-monitoring.md:3]: Missed this one. Resolving now.

File: modules/eks-monitoring/variables.tf [hunk 4]
```diff
@@ -552,6 +552,26 @@ variable "enable_adotcollector_metrics" {
   default     = true
 }
 
+variable "enable_gpu_monitoring" {
```
Comment (bonclay7) [inline modules/eks-monitoring/variables.tf:555]: also should we be more specific than `gpu` and call it `nvidia`?

Comment (lewinkedrs) [reply modules/eks-monitoring/variables.tf:555]: Yes i think that's a good idea.

**Review by bonclay7 — APPROVED** [review]
LGTM

**Pull request merged**